metricbeat/mb/mb.go

```diff
@@ -78,9 +83,23 @@ type EventsFetcher interface {
 // MetricSet interface requirements, leaving only the Fetch() method to be
 // implemented to have a complete MetricSet implementation.
 type BaseMetricSet struct {
-	name   string
-	module Module
-	host   string
+	name     string
+	module   Module
+	host     string
+	hostData HostData
+}
+
+// HostData contains values parsed from the 'host' configuration. Other
+// configuration data like protocols, usernames, and passwords may also be
+// used to construct this HostData data.
+type HostData struct {
+	URI          string // The full URI that should be used in connections.
+	SanitizedURI string // A sanitized version of the URI without credentials.
+
+	// Parts of the URI.
+	Host     string // The host and possibly port.
```

Comment:
In the future we can perhaps extract also the port in case this could become useful.